src/app/_components/Dashboard.tsx

```diff
@@ -1,114 +1,138 @@
-      timerRef.current = setInterval(() => {
-        if (seconds > 0) {
-          setSeconds(seconds - 1);
-        }
-        if (seconds === 0) {
-          if (minutes === 0) {
-            clearInterval(timerRef.current);
-          } else {
-            setMinutes(minutes - 1);
-            setSeconds(59);
-          }
-        }
-      }, 1000);
-    }
-    return () => clearInterval(timerRef.current);
-  }, [activeTimer, minutes, seconds]);
+	const timeToSecs = (hr: number = 0, min: number, sec: number): number =>
+		hr * 3600 + min * 60 + sec;
 
-  useEffect(() => {
-    setPercentage(Math.abs((minutes / startMin) * 100 - 100));
-  }, [minutes, startMin]);
+	const totalStartSec = useMemo(
+		() => timeToSecs(startHr, startMin, startSec),
+		[startHr, startMin, startSec]
+	);
 
-  const restart = () => {
-    // Clears the interval to stop the timer from updating
-    setActiveTimer(false);
-    setMinutes(startMin);
-    setSeconds(startSec);
-  };
+	const timerRef = useRef<NodeJS.Timeout>();
 
-  const pause = () => {
-    // Clears the interval to stop the timer from updating
-    if (activeTimer) {
-      setActiveTimer(false);
-      clearInterval(timerRef.current);
-    }
-  };
+	useEffect(() => {
+		if (activeTimer) {
+			timerRef.current = setInterval(() => {
+				if (seconds === 0) {
+					if (minutes === 0) {
+						if (hours === 0) {
+							setActiveTimer(false);
+						} else {
+							setHours(hours - 1);
+							setMinutes(59);
+							setSeconds(59);
+						}
+					} else {
+						setMinutes(minutes - 1);
+						setSeconds(59);
+					}
+				} else {
+					setSeconds(seconds - 1);
+				}
+			}, 1000);
+		} else {
+			clearInterval(timerRef.current);
+		}
+		return () => clearInterval(timerRef.current);
+	}, [activeTimer, hours, minutes, seconds]);
 
-  return (
-    <main className="flex min-h-screen flex-col items-center space-y-10 p-24">
-      <div
-        className="radial-progress bg-primary text-white border-4 border-primary"
-        style={
-          {
-            "--value": percentage,
-            "--size": "12rem",
-            "--thickness": "2px",
-          } as CSSProperties
-        }
-        role="progressbar"
-      >
-        <div className="grid grid-flow-col gap-1 text-center auto-cols-max">
-          <div className="flex flex-col text-white">
-            <span className="countdown font-mono text-2xl">
-              <span style={{ "--value": minutes } as CSSProperties}></span>:
-            </span>
-            min
-          </div>
+	useEffect(() => {
+		setPercentage((timeToSecs(hours, minutes, seconds) / totalStartSec) * 100);
+	}, [totalStartSec, hours, minutes, seconds]);
+
+	const restart = () => {
+		// Clears the interval to stop the timer from updating
+		setActiveTimer(false);
+		setHours(startHr);
+		setMinutes(startMin);
+		setSeconds(startSec);
+	};
+
+	const pause = () => {
+		// Clears the interval to stop the timer from updating
+		setActiveTimer(false);
+	};
```

Comment:
We need `clearInterval(timerRef.current)` here, toggling activeTimer won't clear the current interval in the useEffect